src/generators/AbstractGenerator.ts

```diff
@@ -41,6 +43,25 @@ export const defaultGeneratorOptions: CommonGeneratorOptions = {
   }
 };
 
+export interface AbstractGeneratorRenderArgs<
+  Options extends CommonGeneratorOptions,
+  ConstrainedModel = ConstrainedMetaModel
+> {
+  constrainedModel: ConstrainedModel;
+  inputModel: InputMetaModel;
+  options?: DeepPartial<Options>;
+}
+
+export interface AbstractGeneratorRenderCompleteModelArgs<
+  Options extends CommonGeneratorOptions,
+  RenderCompleteModelOptions
+> {
+  constrainedModel: ConstrainedMetaModel;
+  inputModel: InputMetaModel;
+  completeOptions: Partial<RenderCompleteModelOptions>;
+  options?: DeepPartial<Options>;
+}
```

Comment:
Any reason you want to break the options, or just left over from the old solution?

Reply:
I was thinking about it after we agreed to change the initial PR. I decided to keep it like this because I think that this approach is better than what we have, but I can remove it in case you disagree.

Reply:
If you can easily re-commit these changes to a separate branch and do a separate PR I would appreciate it 🙏 

It's cluttering the changes of what this PR actually tries to solve, including the discussions will go into multiple directions 😅

Reply:
Removed those changes. I'll submit another PR with those changes after this is merged.